### ISSUE TITLE
Incompatible with Gnome shell 3.34

### ISSUE THREAD

**Issue by hedayat** [projecthamster/shell-extension#315]
Doesn't show the current activities and list of all today activities in gnome shell 3.34

Errors like this can be seen in logs:

```
سپتامبر 22 09:27:10 hv gnome-shell[1876]: JS ERROR: Exception in callback for signal: open-state-changed: TypeError: this.factsBox.focus is not a function
                                                 _init/<@/home/hedayat/.local/share/gnome-shell/extensions/contact@projecthamster.org/widgets/panelWidget.js:130:21
                                                 _emit@resource:///org/gnome/gjs/modules/signals.js:135:27
                                                 open@resource:///org/gnome/shell/ui/popupMenu.js:883:9
                                                 toggle@resource:///org/gnome/shell/ui/popupMenu.js:761:13
                                                 _onEvent@resource:///org/gnome/shell/ui/panelMenu.js:139:13
```

**Comment by jimmytheneutrino** [comment]
@hedayat It seems that at https://github.com/projecthamster/hamster-shell-extension/issues/307 everybody liked the idea of you being a maintainer.
What do you think, could you update your fork with https://github.com/projecthamster/hamster-shell-extension/pull/312, so that new PRs, if they appear, could be sent directly there?

**Comment by hedayat** [comment]
> @hedayat It seems that at #307 everybody liked the idea of you being a maintainer.
> What do you think, could you update your fork with #312, so that new PRs, if they appear, could be sent directly there?

Oh I just forgot about my suggestion :P Well, OK, I'll try to do it soon. Although the idea of an Argos based extension is probably better; but I'm not sure if it can provide all the features of the current extension.

**Comment by mwilck** [comment]
The title of this issue is misleading. It should be "GNOME shell 3.34 incompatible with hamster extension". GNOME core devs, I truly love you. Sorry, I couldn't help it.

@hedayat, which code base have you been testing with? Have you tried the fixes from #307 already?

**Comment by hedayat** [comment]
Yes, I was using 3.32 before; therefore I should have that branch! I'll recheck again but that should be it.

**Comment by mwilck** [comment]
FTR, openSUSE doesn't have GNOME 3.34 yet so I'm not on board so far.

**Comment by hedayat** [comment]
OK, just checked again. No it is a new incompatibility.

**Comment by mwilck** [comment]
I'm now in the boat too, as GNOME 3.34 has hit openSUSE :-)

Let's note that the extension is not totally broken this time. It shows up in the top bar. The text entry field works, so do the buttons for opening the hamster GUI. It just doesn't show the list of recent activities.

**Comment by mwilck** [comment]
Please check https://github.com/mwilck/hamster-shell-extension/pull/1. It works for me with GNOME 3.34. You can also pull directly from https://github.com/mwilck/hamster-shell-extension (master or GNOME-3.34 branch). The changes are small; the code should still work on 3.32 as well, I suppose (can't test easily).

**Comment by hedayat** [comment]
Thanks @mwilck, it fixes the problem with 3.34! And I think your repo can be considered the active fork of this extension. :) I see you are more suited to maintain it than me.

**Comment by mwilck** [comment]
Thanks :-) I guess it's just a matter of taking time to try and fix it. My JS knowledge is truly tiny, I basically just fix the stuff by googling and comparing our code to other people's.

**Comment by hedayat** [comment]
Well, that is not less than what I can do :) And apparently more, as you solved it first :)

**Comment by DirkHoffmann** [comment]
Is this issue still open? I do not see how to get the extension working on gnome shell 3.32 (which seems to have been the initial title), which is the latest available on Fedora 30 today.

**Comment by hedayat** [comment]
@DirkHoffmann  Check this fork: https://github.com/projecthamster/hamster-shell-extension/issues/315#issuecomment-545006053

The main repo is not actively maintained.

**Comment by hedayat** [comment]
If it didn't work for Gnome shell 3.32, use this one:
https://github.com/projecthamster/hamster-shell-extension/pull/312

**Comment by DirkHoffmann** [comment]
Thank you for your quick answer, @hedayat. 
I cloned @mwilck's master (https://github.com/mwilck/hamster-shell-extension) as you suggested. It still yields 
![image](https://user-images.githubusercontent.com/1778649/74600655-dc528980-5094-11ea-95bc-b5fd3695cadd.png)
after compilation, installation and restart of gnome-shell 3.32.2 on Fedora 30. It should be compatible with hamster 2.0.0, right?
Is there a way to see more debug output?

I will try some other tags, as much as my time allows.

**Comment by DirkHoffmann** [comment]
Lesson learned: The branches [GNOME-3.32](https://github.com/mwilck/hamster-shell-extension/tree/GNOME-3.32) and [GNOME-3.34](https://github.com/mwilck/hamster-shell-extension/tree/GNOME-3.34) are not compatible.
As I cannot (need not to) test 3.34 for now, I will stick to #301.
Anyway, thank you for your help, @mwilck and @hedayat. I hope the owners of this project become reasonable and allow you to integrated your PRs soon.

**Comment by ederag** [comment]
@DirkHoffmann the shell-extension is looking for a maintainer, actually:
https://github.com/projecthamster/hamster-shell-extension/issues/315#issuecomment-538941523
https://github.com/projecthamster/hamster-shell-extension/issues/307#issuecomment-498342673

**Comment by DirkHoffmann** [comment]
> @DirkHoffmann the shell-extension is looking for a maintainer, actually:
> [#315 (comment)](https://github.com/projecthamster/hamster-shell-extension/issues/315#issuecomment-538941523)
> [#307 (comment)](https://github.com/projecthamster/hamster-shell-extension/issues/307#issuecomment-498342673)

Thank you for your hint, @ederag? But what is your point? I understood from the two comments that you point, that there are two candidates. (They are active and helped me solving my problem by the way. I am totally happy.) However, one of them detains a fork, which is one year and at least two versions ahead of the official one. 
It would be time for the managers of this repo to give him permissions to or implement pull requests, wouldn't it?

Anything I can do myself to bring things in motion?

**Comment by ederag** [comment]
@DirkHoffmann The issue is *more likely* that noone stepped forward and asked the organization maintainer (@elbenfreund) by mail as I recommended in the second link (https://github.com/projecthamster/hamster-shell-extension/issues/307#issuecomment-498342673). 
I have only the rights to commit in https://github.com/projecthamster/hamster,
not even member of the [organization](https://github.com/orgs/projecthamster/people) yet.

**Comment by ederag** [comment]
@hedayat 
As @jimmytheneutrino said in https://github.com/projecthamster/hamster-shell-extension/issues/315#issuecomment-538941523
> everybody liked the idea of you being a maintainer

That would be great for the project as a whole, IMO.

**Comment by DirkHoffmann** [comment]
All right. We are hijacking this ticket for things that should be settled otherwise.
For the records, I sent an email to all people involved in my opinion. Hope it works without too many further delays. Let's keep fingers crossed!

**Comment by ederag** [comment]
@DirkHoffmann you went way beyond your mandate.

**Comment by mwilck** [comment]
> Lesson learned: The branches [GNOME-3.32](https://github.com/mwilck/hamster-shell-extension/tree/GNOME-3.32) and [GNOME-3.34](https://github.com/mwilck/hamster-shell-extension/tree/GNOME-3.34) are not compatible.

Could you provide some more info, e.g. `journalctl --user -b` output, from running the 3.34 code in under 3.32?

**Comment by mwilck** [comment]
@DirkHoffmann, could you try if reverting 69af56af fixes the issue on GNOME 3.32?

**Comment by DirkHoffmann** [comment]
> @DirkHoffmann, could you try if reverting [69af56a](https://github.com/projecthamster/hamster-shell-extension/commit/69af56aff2e49cf8888f70a761bfa281ad48ef95) fixes the issue on GNOME 3.32?

Not quickly, but let me assign this to the two of us.

**Comment by DirkHoffmann** [comment]
> @DirkHoffmann, could you try if reverting [69af56a](https://github.com/projecthamster/hamster-shell-extension/commit/69af56aff2e49cf8888f70a761bfa281ad48ef95) fixes the issue on GNOME 3.32?

> Could you provide some more info, e.g. `journalctl --user -b` output, from running the 3.34 code in under 3.32?

But wait, this should be handled in #301.

**Comment by DirkHoffmann** [comment]
I close this discussion in favour of #316 (and beforementioned #301), unless someone else wants to speak up here.
